dateutil/rrule.py

```diff
@@ -1501,6 +1503,11 @@ def _parse_rfc(self, s,
         if compatible:
             forceset = True
             unfold = True
+
+        TZID_NAMES = dict(map(
```

Comment:
Both `dict` and `map` are very slow compared to a dictionary comprehension in this case:

```python
TZID_NAMES = {x.upper(): x
              for x in re.findall('TZID=(?P<name>[^:]+):', s)}
```

That said, I don't *love* this approach (though it *may* be the best option, unfortunately). I'd fix the comprehension part but don't worry about the regex part - we can fix that with a bigger refactor later I think.